wicketd/src/http_entrypoints.rs

```diff
@@ -834,21 +849,27 @@ async fn get_location(
     }))
 }
 
-/// An endpoint to start updating a sled.
+/// An endpoint to start updating one or more sleds, switches and PSCs.
 #[endpoint {
     method = POST,
-    path = "/update/{type}/{slot}",
+    path = "/update",
 }]
 async fn post_start_update(
     rqctx: RequestContext<ServerContext>,
-    target: Path<SpIdentifier>,
-    opts: TypedBody<StartUpdateOptions>,
+    params: TypedBody<StartUpdateParams>,
 ) -> Result<HttpResponseUpdatedNoContent, HttpError> {
     let log = &rqctx.log;
     let rqctx = rqctx.context();
-    let target = target.into_inner();
+    let params = params.into_inner();
+
+    if params.targets.is_empty() {
+        return Err(HttpError::for_bad_request(
+            None,
+            "No update targets specified".into(),
+        ));
+    }
 
-    // Can we update the target SP? We refuse to update if:
+    // Can we update the target SPs? We refuse to update if:
```

Comment:
Tiny nit since the list refers to single SPs:

```suggestion
    // Can we update the target SPs? We refuse to update if for any target SP:
```

Reply:
Done, thanks!